internal/pkg/utils/utils.go

```diff
@@ -45,3 +46,44 @@ func FieldName(fa *ssa.FieldAddr) string {
 	}
 	return st.Field(fa.Field).Name()
 }
+
+// DecompoeType returns the path, typename, and indicators for if the Type is Named or an Interface
+// Returns empty strings if the type is not *types.Named
+func DecomposeType(t types.Type) (path, name string) {
+	n, ok := t.(*types.Named)
+	if !ok {
+		return
+	}
+
+	if pkg := n.Obj().Pkg(); pkg != nil {
+		path = pkg.Path()
+	}
+
+	return path, n.Obj().Name()
+}
+
+func UnqualifiedName(v *types.Var) string {
+	packageQualifiedName := v.Type().String()
+	dotPos := strings.LastIndexByte(packageQualifiedName, '.')
+	if dotPos == -1 {
+		return packageQualifiedName
+	}
+	return packageQualifiedName[dotPos+1:]
+}
+
+// DecomposeFunction returns the path, receiver, and name strings of a ssa.Function.
+// For functions that have no receiver, returns an empty string for recv.
+// If f is nil, returns empty strings for all return values.
+func DecomposeFunction(f *ssa.Function) (path, recv, name string) {
+	if f == nil {
+		return
+	}
+
+	path = f.Pkg.Pkg.Path()
```

Comment:
This looks potentially dangerous to me, given this comment in `ssa.go`, line 306:
```go
	Pkg       *Package      // enclosing package; nil for shared funcs (wrappers and error.Error)
```
(in the definition for `ssa.Function`)

Reply:
Safety check added.